jaq-json/src/lib.rs

```diff
@@ -765,9 +765,14 @@ impl core::ops::Div for Val {
 impl core::ops::Rem for Val {
     type Output = ValR;
     fn rem(self, rhs: Self) -> Self::Output {
-        use Val::Int;
+        use Val::{Float, Int, Num};
         match (self, rhs) {
             (Int(x), Int(y)) if y != 0 => Ok(Int(x % y)),
+            (Float(f), Int(i)) if i != 0 => Ok(Int(f as isize % i)),
+            (Int(i), Float(f)) if f != 0.0 => Ok(Int(i % f as isize)),
+            (Float(x), Float(y)) if y != 0.0 => Ok(Int(x as isize % y as isize)),
+            (Num(n), r) => Self::from_dec_str(&n) % r,
+            (l, Num(n)) => l % Self::from_dec_str(&n),
```

Comment:
As i understand these will end up in the operator overload `rem` again?

Reply:
I'm sorry, I do not understand your question.

Reply:
Anyway, to remain consistent with division, I think that we should not check for `rhs != 0` *if we use the RHS as floating-point number*. This is the case for all constellations in this function except when both operands are integers.

You may ask why we do not just convert all integers to floats like we did in division. The answer is: because for any two integers `l` and `r` (where `r != 0`), `l % r` is always integer, whereas `l / r` may yield non-integer results. That's why we should not convert integers `l` and `r` to floats when we calculate `l % r`, but why we should do it when we calculate `l / r`.

For reference: In Rust, `1.0 % 0.0` is NaN, whereas `1 % 0` panics.

TL;DR: If you remove the last three checks for equality with `0`, I think that this PR is good to merge.

Reply:
It was mostly me not encountered rust operator overloading before. Ah thanks for explaining about floats and rem, now a bit clearer 👍 i've pushed an updated for this